modules/getComponents.js

```diff
@@ -11,35 +9,9 @@ function getComponentsForRoute(nextState, route, callback) {
   const getComponent = route.getComponent || route.getComponents
   if (!getComponent) {
     callback()
-    return
-  }
-
-  const { location } = nextState
-  let nextStateWithLocation
-
-  if (__DEV__ && canUseMembrane) {
-    nextStateWithLocation = { ...nextState }
-
-    // I don't use deprecateObjectProperties here because I want to keep the
-    // same code path between development and production, in that we just
-    // assign extra properties to the copy of the state object in both cases.
-    for (const prop in location) {
-      if (!Object.prototype.hasOwnProperty.call(location, prop)) {
-        continue
-      }
-
-      Object.defineProperty(nextStateWithLocation, prop, {
-        get() {
-          warning(false, 'Accessing location properties from the first argument to `getComponent` and `getComponents` is deprecated. That argument is now the router state (`nextState`) rather than the location. To access the location, use `nextState.location`.')
-          return location[prop]
-        }
-      })
-    }
   } else {
-    nextStateWithLocation = { ...nextState, ...location }
+    getComponent.call(route, nextState, callback)
```

Comment:
Nit: this shouldn't be in an `else`. A number of linting rules prohibit "pointless" else clauses per https://github.com/airbnb/javascript/blob/a9db06e1ddbefe3342a61b649c104420d82ba37d/packages/eslint-config-airbnb/rules/best-practices.js#L39. Just put this outside the `if` block as in https://github.com/reactjs/react-router/pull/3348.